tests/unit/regression/krrnystrom_unittest.cc

```diff
@@ -0,0 +1,133 @@
+* ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+* (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+* SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*
+* The views and conclusions contained in the software and documentation are those
+* of the authors and should not be interpreted as representing official policies,
+* either expressed or implied, of the Shogun Development Team.
+*/
+
+#include <shogun/features/DenseFeatures.h>
+#include <shogun/labels/RegressionLabels.h>
+#include <shogun/kernel/GaussianKernel.h>
+#include <shogun/regression/KRRNystrom.h>
+#include <shogun/regression/KernelRidgeRegression.h>
+#include <gtest/gtest.h>
+
+using namespace shogun;
+
+/**
+ * Test that the expected interface is available.
+ */
+TEST(KRRNystrom, test_interface)
+{
+
+	/* Use default constructor */
+	CKRRNystrom krr;
+
+	/* Test setters */
+	krr.set_tau(0.1);
+	krr.set_m(100);
+}
+
+TEST(KRRNystrom, test_get_name)
+{
+	CKRRNystrom krr;
+	EXPECT_EQ(krr.get_name(), "KRRNystrom");
+}
+
+TEST(KRRNystrom, test_get_classifier_type)
+{
+	CKRRNystrom krr;
+	EXPECT_EQ(krr.get_classifier_type(), CT_KERNELRIDGEREGRESSION);
+}
+
+/**
+ * Test the main algorithm by comparison of the alphas and predictions
+ */
+TEST(KRRNystrom, compare_to_KRR)
+{
+	/* data matrix dimensions */
+	index_t num_vectors=10;
+	index_t num_features=1;
+
+	/* training label data */
+	SGVector<float64_t> lab(num_vectors);
+
+	/* fill data matrix and labels */
+	SGMatrix<float64_t> train_dat(num_features, num_vectors);
+	SGMatrix<float64_t> test_dat(num_features, num_vectors);
+	for (index_t i=0; i<num_vectors; ++i)
+	{
+		/* labels are linear plus noise */
+		lab.vector[i]=i+CMath::normal_random(0, 1.0);
+		train_dat.matrix[i]=i;
+		test_dat.matrix[i]=i;
+	}
+
+	/* training features */
+	CDenseFeatures<float64_t>* features=
+			new CDenseFeatures<float64_t>(train_dat);
+	CDenseFeatures<float64_t>* features_krr=
+			new CDenseFeatures<float64_t>(train_dat);
+
+	/* testing features */
+	CDenseFeatures<float64_t>* test_features=
+			new CDenseFeatures<float64_t>(test_dat);
+
+	/* training labels */
+	CRegressionLabels* labels=new CRegressionLabels(lab);
+	CRegressionLabels* labels_krr=new CRegressionLabels(lab);
+
+	/* kernel */
+	CGaussianKernel* kernel=new CGaussianKernel(features, features, 10, 0.5);
+	CGaussianKernel* kernel_krr=new CGaussianKernel(features, features_krr, 10, 0.5);
+
+	/* kernel ridge regression and the nystrom approximation */
+	float64_t tau=0.0001;
+	CKRRNystrom* nystrom=new CKRRNystrom(tau, num_vectors, kernel, labels);
+	CKernelRidgeRegression* krr=new CKernelRidgeRegression(tau, kernel_krr, labels_krr);
+
+	nystrom->train();
+	krr->train();
+
+	SGVector<float64_t> alphas=nystrom->get_alphas();
+	SGVector<float64_t> alphas_krr=krr->get_alphas();
+
+	for (index_t i=0; i<num_vectors; ++i) {
+		EXPECT_NEAR(alphas[i], alphas_krr[i], 1E-1);
```

Comment:
this should be 1e-5 at least in my experience if all of the matrix is used (only difference should be the numerics of using the SVD to invert the kernel matrix)